examples/cmd/snapshotting/README.md

```diff
@@ -0,0 +1,59 @@
+# Snapshotting demo
+
+This example shows snapshotting in action by sending a marker to a VM via a running process (in this case `sleep 422`), snapshotting the VM, closing it, loading and starting a new machine via the same snapshot, and checking for the marker.
+
+## Running the test
+
+Run this test by first running
+
+```
+make all
+```
+
+followed by
+
+```
+sudo -E env PATH=$PATH go run example_demo.go
+```
+
+Alternatively, to do both of the above,
+```
+make run
+```
+
+Note the user PATH variable is different from the root user's PATH variable, hence the need for `-E env PATH=$PATH`.
+
+Upon running, the VM logs will be printed to the console, as well as the IP of the VM. It will then show that it is sending the marker (in our case, `sleep 422`).
+
+Afterwards, the snapshot is created and the machine is terminated. The snapshot files are saved in the snapshotssh folder created in the directory.
+
+Then, a new machine is created, booted with the snapshot that was just taken, and the IP of the VM will once again be printed to the console (which should be the same as the last machine). The output of searching for the marker (in this case `ps -aux | grep "sleep 422"`) is then printed to the console and the user can confirm that the snapshot loaded properly.
+
+Note: Lines 223 and 349 have `fmt.Scanln()` commented out. Uncommenting them pauses the program, allowing you to ssh into the machine in a new terminal window with the following command:
```

Comment:
From a documentation perspective this feels tedious to maintain, but a little more beyond that the commented lines of code are unnecessary.

Reply:
Per what @ginglis13 suggested I made it more of a general case, it still involves a particular code snippet but should be better? What do you think?